server/impl/src/main/java/com/walmartlabs/concord/server/org/secret/SecretEntryV2.java

```diff
@@ -0,0 +1,156 @@
+package com.walmartlabs.concord.server.org.secret;
+
+/*-
+ * *****
+ * Concord
+ * -----
+ * Copyright (C) 2017 - 2018 Walmart Inc.
+ * -----
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =====
+ */
+
+import com.fasterxml.jackson.annotation.*;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.walmartlabs.concord.common.secret.SecretEncryptedByType;
+import com.walmartlabs.concord.common.validation.ConcordKey;
+import com.walmartlabs.concord.server.org.EntityOwner;
+import com.walmartlabs.concord.server.org.project.ProjectEntry;
+
+import javax.validation.constraints.NotNull;
+import java.io.Serializable;
+import java.util.*;
+
+@JsonInclude(Include.NON_NULL)
+public class SecretEntryV2 implements Serializable {
```

Comment:
I think that we can add new fields into `SecretEntry` and mark, `project, projectName` as deprecated.
Because `SecretEntry` with project/projectName make no sense...

Reply:
This is to not touch the old api's. If someone using the old api and not expecting these new attribute (projects), its gonna be a problem. Eventually we can deprecate the V1.